### DISCUSSION TITLE
How to animate TabPanel?

### DISCUSSION THREAD

**Discussion by Akbar-ATQ** [chakra-ui/chakra-ui#5085] [Q&A] [answered]
I want to animate TabPanel when moving between Tab with framer motion, I know how to turn the TabPanel into motion component by using motion(TabPanel), but the animation only appear in the first load, after that there is no animation.
Also how to use AnimatePresence to animate TabPanel exit animation? because TabPanels can only have TabPanel as a child, I can't wrap the MotionTabPanel in AnimatePresence.

**Comment by Akbar-ATQ** [comment]
Also, when the you wrap TabPanel in other element like Flex, the panel won't show. Meanwhile when you wrap Tab in other element like Flex, nothing change. It still work normally even though the docs say TabList should only have Tab as children.
Is that bug or intended behavior?

**Comment by malashevskyi** [comment, marked as answer]
[codesandbox example](https://codesandbox.io/s/animate-chakra-ui-tabs-cr6zg)
1. set 'isLazy' on Tabs
```javascript
<Tabs isLazy={true}> </Tabs>
```
2. then
```javascript
<TabPanel as={AnimatePresence} > /* your motion tag */ </TabPnel>
```
3. you need to set absolute style for tab panel content, otherwise it will jump during changing
```javascript
 <Box
      as={motion.div}
      pos="absolute" > </Box>
```
4. So, tabPanels is collapsed as it position absolute, set height of current tab panel
```javascript
// changeHeight prop form state [height, setHeight] = <TabPanels h={height}>... <TabPanelContent changeHeight={setHeight}>
const TabPanelContent = ({ children, changeHeight }) => { 
  const currentPanelRef = useRef(document.createElement('div'))
  useEffect(() => {
    changeHeight(currentPanelRef.current.clientHeight + 'px')
  })
  return (
    <Box
      as={motion.div}
      pos="absolute"
      ref={currentPanelRef}
    >
      {children}
    </Box>
  )
}
```

**Reply by Akbar-ATQ** [reply]
Thanks, its work.

**Reply by tannguyencse19** [reply]
I don't need `<AnimatePresence />` to work, so it solves `jump during changing` too. 
_P/S: Still fix `jump during changing` using `pos="absolute`_

Also, it cause `tab-panel` HTML to be removed if not active, which will cause error in mapping key between `Tab` and `TabPanel` .

So the code will be simplify to:
```jsx
<TabPanels>
   {data.map((tab) => (
      <TabPanel key={`tab-${tab.name}`}>
          <motion.div> // <-- Notes: Can't merge this to <TabPanel />`like this: <TabPanel as={motion.div} />
             {tab.content}
           </motion.div>
      </TabPanel>
   )}
</TabPanels>
```

Still, It's a good trick if anyone want to force animation.

**Comment by Fazliddin-04** [comment]
Hi @malashevskyi, thanks for your answer. But there is something wrong with the codesandbox example.
